File: dodian-backend/dodian-server/src/main/java/net/dodian/uber/game/model/object/Object.java
package net.dodian.uber.game.model.object;

public class Object {
    public int id, x, y, z, type, face, oldId;
    private java.lang.Object attachment = null;

    public Object(int id, int x, int y, int z, int type) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.z = z;
        this.type = type;
    }

    public Object(int id, int x, int y, int z, int type, int face) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.z = z;
        this.type = type;
        this.face = face;
    }

    public Object(int id, int x, int y, int z, int type, int face, int oldId) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.z = z;
        this.type = type;
        this.face = face;
        this.oldId = oldId;
    }

    public void setAttachment(java.lang.Object o) {
        this.attachment = o;
    }

    /**
     * @return the attachment
     */
    public java.lang.Object getAttachment() {
        return attachment;
    }
}

File: game-server/src/main/kotlin/net/dodian/uber/game/model/object/Object.kt
package net.dodian.uber.game.model.`object`

class Object @JvmOverloads constructor(
    @JvmField var id: Int,
    @JvmField var x: Int,
    @JvmField var y: Int,
    @JvmField var z: Int,
    @JvmField var type: Int,
    @JvmField var face: Int = 0,
    @JvmField var oldId: Int = 0,
) {
    private var attachment: Any? = null

    fun setAttachment(attachment: Any?) {
        this.attachment = attachment
    }

    fun getAttachment(): Any? = attachment
}
